modules/images/dominant-color/class-dominant-color-image-editor-gd.php

```diff
@@ -35,11 +35,7 @@ public function get_dominant_color() {
 		$shorted_image = imagecreatetruecolor( 1, 1 );
 		imagecopyresampled( $shorted_image, $this->image, 0, 0, 0, 0, 1, 1, imagesx( $this->image ), imagesy( $this->image ) );
 
-		$rgb = imagecolorat( $shorted_image, 0, 0 );
-		$r   = ( $rgb >> 16 ) & 0xFF;
-		$g   = ( $rgb >> 8 ) & 0xFF;
-		$b   = $rgb & 0xFF;
-		return sprintf( '%02x%02x%02x', $r, $g, $b );
+		return str_pad( dechex( imagecolorat( $shorted_image, 0, 0 ) ), 6, '0', STR_PAD_LEFT );
```

Comment:
Let's go back to this. 

```
sprintf( '%02x%02x%02x', $color['r'], $color['g'], $color['b'] );
```

It is pretty clear what it does. It uses red, green and blue to generate a hex colour. This is used elsewhere and works just fine.